### ISSUE TITLE
Table header is too aggressive when first cell is specified with AsciiDoc style

### ISSUE THREAD

**Issue by graphitefriction** [asciidoctor/asciidoctor#644]
When the first cell is prefixed with `a` and is directly beneath the table delimiter, the first row is styled as a header even though the `header` attribute is not set in the attribute list or implicitly.

This code renders incorrectly (i.e. row is formatted as a header and AsciiDoc styles are nullified).

```
[cols="2"]
|===
a|This cell is prefixed with an +a+ so the following list is rendered with the AsciiDoc list element styles.

* List item 1
* List item 2
* List item 3
|This cell *is not* prefixed with an +a+ so the following list is not rendered with the AsciiDoc list element styles.

* List item 1
* List item 2
* List item 3

a|This cell is prefixed with an +a+ so the following paragraph is rendered with the +lead+ style.

[.lead]
I am a paragraph styled with the lead attribute.
|This cell *is not* prefixed with an +a+ so the following paragraph is not rendered with the +lead+ style.

[.lead]
I am a paragraph styled with the lead attribute.

|===
```

**Work Around**

Insert a blank line between the table delimiter and the first cell.

This code works correctly (i.e. the first row is not formatted as a header)

```
[cols="2"]
|===

a|This cell is prefixed with an +a+ so the following list is rendered with the AsciiDoc list element styles.

* List item 1
* List item 2
* List item 3
|This cell *is not* prefixed with an +a+ so the following list is not rendered with the AsciiDoc list element styles.

* List item 1
* List item 2
* List item 3

a|This cell is prefixed with an +a+ so the following paragraph is rendered with the +lead+ style.

[.lead]
I am a paragraph styled with the lead attribute.
|This cell *is not* prefixed with an +a+ so the following paragraph is not rendered with the +lead+ style.

[.lead]
I am a paragraph styled with the lead attribute.

|===
```

**Comment by mojavelinux** [comment]
A simple way to fix this is to disable the implicit header behavior if any cell on the first line has a cell style. Cell styles are ignored for cells in the header row, so we know by that it's not a header row.

**Comment by mojavelinux** [comment]
Actually, this is the same issue as #1284. What we need to look for is a cell still being open after the first line. In that case, we need to reverse the decision to make an implicit header row.

**Comment by mojavelinux** [comment]
Resolved by fix for #1284.
